client/verta/verta/pipeline/_registered_pipeline.py

```diff
@@ -0,0 +1,174 @@
+# -*- coding: utf-8 -*-
+
+import json
+import copy
+from typing import Any, Dict, Optional
+
+from verta._internal_utils._utils import Configuration, Connection
+from verta.endpoint.resources import Resources
+from verta.pipeline import PipelineGraph
+from verta.registry.entities import RegisteredModelVersion
+
+
+class RegisteredPipeline:
+    """
+    Object representing a version of a registered inference pipeline.
+    There should not be a need to instantiate this class directly; please use
+    :meth:`Client.create_registered_pipeline() <verta.Client.create_registered_piepline>`
+    for creating a new pipeline, or
+    :meth:`Client.get-registered_pipeline() <verta.Client.get_registered_piepline>`
+    for fetching existing pipelines.
+
+    Attributes
+    ----------
+        name: str
+            Name of this pipeline.
+        id: int
+            Auto-assigned ID of this Pipeline.
+        pipeline_graph: :class:`~verta.pipeline.PipelineGraph`
+            PipelineGraph object containing all possible steps in the Pipline.
+    """
+
+    def __init__(
+        self,
+        registered_model_version: RegisteredModelVersion,
+        pipeline_graph: PipelineGraph,
+    ):
+        """
+        Create a Pipeline instance from an existing RegisteredModelVersion object
+        and the provided pipeline graph. Name and ID are captured once upon creation
+        to avoid additional HTTP calls to refresh the cache of the RMV, because
+        pipelines are immutable.
+        """
+        self._registered_model_version = registered_model_version
+        self._name = self._registered_model_version.name
+        self._id = self._registered_model_version.id
+        self._pipeline_graph = pipeline_graph
+
+    def __repr__(self):
+        return "\n".join(
+            (
+                "RegisteredPipeline:",
+                f"pipeline name: {self.name}",
+                f"pipeline id: {self.id}",
+                f"\n{self._pipeline_graph}",
+            )
+        )
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def id(self):
+        return self._id
+
+    @property
+    def pipeline_graph(self):
+        return self._pipeline_graph
+
+    def copy_graph(self) -> PipelineGraph:
+        """
+        Return a shallow copy of the PipelineGraph of this pipeline. RegisteredPipeline
+        objects are immutable once registered with Verta. This function returns
+        a PipelineGraph object that can be modified and used to create and register
+        a new RegisteredPipeline.
+        """
+        return copy.copy(self.pipeline_graph)
```

Comment:
I'd recommend a deep copy, unless it's prohibitively slower. If a user mutates a step in the graph "copy"

```python
pipeline = client.get_pipeline("foo")
new_graph = pipeline.copy_graph()
new_graph.steps[1].set_model_version(new_rmv)
```

It'll affect the original `pipeline.pipeline_graph` too.

Reply:
Done